plumgrid_playbooks/roles/plumgrid_common/tasks/main.yml

```diff
@@ -14,24 +14,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# Remove sources file if exists
 - name: Removing pre-exiting sources.list
-  file: path=/etc/apt/sources.list.d/plumgrid.list state=absent
-
-# Create a PLUMgrid sources.list
-- name: Create PLUMgrid sources.list
-  lineinfile:
-    dest: /etc/apt/sources.list.d/plumgrid.list
-    line: "deb {{ plumgrid_repo }}/plumgrid plumgrid {{ plumgrid_component }}"
-    state: present
-    create: yes
+  file:
+    path: "/etc/apt/sources.list.d/plumgrid.list"
+    state: absent
 
-# Create a PLUMgrid sources.list
-- name: Add PLUMgrid-images to repo
-  lineinfile:
-    dest: /etc/apt/sources.list.d/plumgrid.list
-    line: "deb {{ plumgrid_repo }}/plumgrid-images plumgrid {{ plumgrid_component }}"
+- name: Add PLUMgrid and PLUMgrid-images repo
+  apt_repository:
```

Comment:
Any special reason to not use the `apt_repository` module? it's much easy, clean and recommended way